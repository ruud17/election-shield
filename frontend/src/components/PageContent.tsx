import { FC, useState, ChangeEvent } from "react";
import { Container } from "react-bootstrap";
import VoterDetailsForm from "./VoterDetailsForm";
import CandidatesSelection from "./CandidatesSelection";
import SubmitVote from "./SubmitVote";
import { VoterDetails } from "../utils/interfaces";
import { initialVoterDetailsData } from "../utils/defaultStates";
import { Candidates } from "../utils/enum";

const PageContent: FC = () => {
  const [voterDetails, setVoterDetails] = useState<VoterDetails>(
    initialVoterDetailsData
  );

  const [selectedCandidate, setSelectedCandidate] = useState<string>(
    Candidates.DonaldTrump
  );

  const [isTermsCheckboxConfirmed, setIsTermsCheckboxConfirmed] =
    useState<boolean>(false);

  const handleVoterDetailChange = (event: ChangeEvent<HTMLInputElement>) => {
    const { name, value } = event.target;
    setVoterDetails({
      ...voterDetails,
      [name]: value,
    });
  };

  const handleCandidateSelectionChange = (
    event: ChangeEvent<HTMLInputElement>
  ) => {
    setSelectedCandidate(event.target.value);
  };

  const handleTermsOfServiceOptionConfirmation = () => {
    setIsTermsCheckboxConfirmed(!isTermsCheckboxConfirmed);
  };

  return (
    <Container className="page-content">
      <VoterDetailsForm
        voterDetails={voterDetails}
        handleInputChange={handleVoterDetailChange}
      />
      <CandidatesSelection
        selectedCandidate={selectedCandidate}
        handleOptionChange={handleCandidateSelectionChange}
      />
      <SubmitVote
        isTermsCheckboxConfirmed={isTermsCheckboxConfirmed}
        handleChange={handleTermsOfServiceOptionConfirmation}
      />
    </Container>
  );
};

export default PageContent;
