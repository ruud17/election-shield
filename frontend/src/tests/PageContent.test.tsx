import renderer from "react-test-renderer";
import PageContent from "../components/PageContent";

test("renders correctly", () => {
  const tree = renderer.create(<PageContent />).toJSON();

  expect(tree).toMatchSnapshot();
});
