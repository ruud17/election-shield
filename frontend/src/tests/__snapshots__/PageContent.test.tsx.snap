// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="page-content container"
>
  <form
    className="voter-details-form"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="mb-3 form-floating"
        >
          <input
            className="form-control form-control-sm"
            id="firstName"
            name="firstName"
            onChange={[Function]}
            placeholder="First Name"
            type="text"
            value=""
          />
          <div
            className="invalid-feedback"
          />
          <label
            htmlFor="firstName"
          >
            First Name
          </label>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="mb-3 form-floating"
        >
          <input
            className="form-control form-control-sm"
            id="lastName"
            name="lastName"
            onChange={[Function]}
            placeholder="Last Name"
            type="text"
            value=""
          />
          <div
            className="invalid-feedback"
          />
          <label
            htmlFor="lastName"
          >
            Last Name
          </label>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="mb-3 form-floating"
          >
            <input
              className="form-control form-control-sm"
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="email"
              value=""
            />
            <div
              className="invalid-feedback"
            />
            <label
              htmlFor="email"
            >
              Email
            </label>
          </div>
        </div>
        <div
          className="col"
        />
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="mb-3 form-floating"
          >
            <input
              aria-describedby="idHelpBlock"
              className="form-control form-control-sm"
              id="identity_id"
              name="identity_id"
              onChange={[Function]}
              placeholder="Identity Id"
              type="string"
              value=""
            />
            <small
              className="form-text text-muted"
              id="idHelpBlock"
            >
              Please enter your identity id from your ID card
            </small>
            <div
              className="invalid-feedback"
            />
            <label
              htmlFor="identity_id"
            >
              Identity Id
            </label>
          </div>
        </div>
      </div>
    </div>
  </form>
  <div
    className="candidate row"
  >
    <div
      className="fade alert alert-info show"
      role="alert"
    >
      Select the presidential candidate you'd like to vote for.
    </div>
    <div>
      <div
        className="form-check"
      >
        <input
          checked={true}
          className="form-check-input"
          disabled={false}
          onChange={[Function]}
          type="radio"
          value="Donald Trump"
        />
        <label
          className="form-check-label"
          title=""
        >
          Donald Trump
        </label>
      </div>
      <div
        className="form-check"
      >
        <input
          checked={false}
          className="form-check-input"
          disabled={false}
          onChange={[Function]}
          type="radio"
          value="Joe Biden"
        />
        <label
          className="form-check-label"
          title=""
        >
          Joe Biden
        </label>
      </div>
    </div>
  </div>
  <div
    className="submit-vote row"
  >
    <div
      className="col"
    >
      <div
        className="form-check"
      >
        <input
          checked={false}
          className="form-check-input"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="form-check-label"
        >
          Accept Terms of Service and Data Collection Consent
        </label>
      </div>
    </div>
    <div
      className="d-flex justify-content-end col"
    >
      <button
        className="btn btn-primary btn-sm"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </div>
    <div
      className="col"
    >
      <div
        className="row"
      />
    </div>
  </div>
</div>
`;
